fbpic/particles/spin/spin_tracker.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2023, FBPIC contributors
+# Author: Michael J. Quin, Kristjan Poder
+# Scientific supervision: Matteo Tamburini
+# License: 3-Clause-BSD-LBNL
+"""
+This file is part of the Fourier-Bessel Particle-In-Cell code (FB-PIC)
+It defines the structure and methods associated with particle spin
+tracking.
+"""
+import numpy as np
+# Check if CUDA is available, then import CUDA functions
+from fbpic.utils.cuda import cuda_installed
+from .numba_methods import push_s_numba, push_s_ioniz_numba
+from .cuda_numba_utils import random_point_sphere_gpu, \
+    random_point_sphere_cpu
+
+if cuda_installed:
+    import cupy
+    from fbpic.utils.cuda import cuda_tpb_bpg_1d
+    from .cuda_methods import push_s_gpu, push_s_ioniz_gpu
+
+
+class SpinTracker(object):
+    """
+    Class that stores and tracks particle spin vector.
+    """
+    def __init__(self, species, dt, sx_m=0., sy_m=0., sz_m=1.,
+                 anom=0., spin_distr='fixed'):
+        """
+        Initialize the SpinTracker class.
+
+        The length of each particle's spin vector is 1, and this
+        does not change during the push. During initialization,
+        the spin vectors will be all set to the same value or
+        randomly generated to have the average values along all
+        axis specified by the user (see spin_distr below).
+
+        .. math::
+            \\frac{d\\boldsymbol{s}}{dt} = (\\boldsymbol{\\Omega}_T +
+             \\boldsymbol{\\Omega}_a) \\boldsymbol{s}
```

Comment:
This is supposed to be a vector product, correct?
```suggestion
             \\boldsymbol{\\Omega}_a) \\times \\boldsymbol{s}
```